gui/control.js

```diff
@@ -17,13 +17,44 @@ const addAccountButton = document.getElementById('add-account-btn');
 window.addEventListener('unload', beforeQuit)
 window.addEventListener('DOMContentLoaded', main)
 
+const loadingMessage = document.getElementById('loading-message');
+const loadingOverlay = document.getElementById('loading-overlay');
+const showOverlay =false;
 
-async function addButtonPressed() {
+function setLoadingOverlayVisibility(showOverlay) {
+  loadingOverlay.style.display = showOverlay ? 'flex' : 'none';
+}
+
+async function runWithLoading(toRun) {
+  try {
+    loadingOverlay.style.display = "flex";
+
+    // Wait for the asynchronous task to complete
+    await toRun();
+  } finally {
+    loadingOverlay.style.display = "none";
+  }
+}
+
+function switchAcc(nickname) {
+  await runWithLoading(async () => {
```

Comment:
will need ot remove await from this line

Reply:
Have already done this in previous commit.